c7n/utils.py

```diff
@@ -169,7 +169,7 @@ def default(self, obj):
         try:
             return json.JSONEncoder.default(self, obj)
         except TypeError:
-            if getattr(obj, '__dict__'):
+            if hasattr(obj, '__dict__'):
```

Comment:
hasattr swallows exceptions, getattr with a default won't

Reply:
will that somehow affect the current exception that i'd be re-raising if __dict__ is not an attribute, then?  so it'd be better to use `if getattr(obj, '__dict__', None) is not None:`?